content/5_Plat/Binary_Jump.mdx

```diff
@@ -129,6 +129,42 @@ int main() {
 
 </CPPSection>
 
+<PySection>
+
+```py
+import math
+
+
+n, q = map(int, input().split())
+
+
+def kth_ancestor(x: int, k: int) -> int:  # return the kth ancestor of x
+	for i in range(LOGMAX):
+		if k & 1 << i:
+			x = up[i][x]
+		if x == -2:
+			break
+	return x
+
+
+LOGMAX = math.ceil(math.log2(n))
+
+up = [[-2 for j in range(n)] for i in range(LOGMAX + 1)]
```

Comment:
-2 is kinda a weird number, use `None` instead

Reply:
It can't be None !! `up[i][j] = up[i - 1][up[i - 1][j]]` . The index can't be None

Reply:
when did I say the index would be none? I meant the -2

Reply:
I don't know what you mean

Reply:
if up[i-1][j] can be None which results in an error, it should result in bad behavior if it's -2 as well

Reply:
Done.